GoogleForCreators/web-stories-wp
# Hotlinking: Allow story poster image to be hotlinked

## Context



## Summary

Allow users without the ability to upload media to set a story poster. Enable hotlinking of story posters. 

## Relevant Technical Choices

Add a new piece of post meta to store an object containing url, width and height for hotlinked image. 

All places that use feature image, add an override to fallback to hotlinked image if one is found. 

Add new poster height / width fields to the the Story object. 

Use story model in more places, so this reflects changes to poster. 

## To-do



## User-facing changes

https://user-images.githubusercontent.com/237508/175333859-a51178a2-b4df-40d2-a4c2-ab56d452dd71.mov

## Testing Instructions






This PR can be tested by following these steps:

1. Enable hotlinking poster experiment. 
2. Go to documents tab. 
3. Click on poster and pencil icon. 
4. See option to Link to file. 
5. Open dialog. 
6. Try linking an image. See error message. 
7. Try linking to https://assets.computingcorner.co.uk/uploads/2022/06/cropped-EbvX20imFmM-3.jpeg
8. See poster image work. 
9. Publish story. 
10. Go to dashboard. See poster image. 
11. Create new post. 
12. Insert web stories block. 
13. Select "Selected Stories" button. 
14. Select a layout style. 
15. Click select button. 
16. See poster image. 

#### Test different sized images
Steps 1-6
Try the following urls. 
https://images.unsplash.com/photo-1656523267493-31b9b2cfdc47?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxlZGl0b3JpYWwtZmVlZHwxfHx8ZW58MHx8fHw%3D&auto=format&fit=crop&w=500&q=60

https://images.unsplash.com/photo-1656523267493-31b9b2cfdc47?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxlZGl0b3JpYWwtZmVlZHwxfHx8ZW58MHx8fHw%3D&auto=format&fit=crop&w=640&q=60

https://images.unsplash.com/photo-1656523267493-31b9b2cfdc47?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxlZGl0b3JpYWwtZmVlZHwxfHx8ZW58MHx8fHw%3D&auto=format&fit=crop&w=500&q=60&h=853

See different error messages. Relating to size. 


## Reviews

### Does this PR have a security-related impact?



### Does this PR change what data or activity we track or use?



### Does this PR have a legal-related impact?



## Checklist




---



Fixes #11536
